fusionfrontend/src/app/factory/components/content/asset-details/asset-charts/asset-charts.component.ts

```diff
@@ -83,51 +85,152 @@ export class AssetChartsComponent implements OnInit, OnChanges, OnDestroy {
+          case '1hour':
+            this.flushData();
+            this.loadHistoricData(this.maxPoints, false, 3600);
+            break;
+          case '1day':
+            this.flushData();
+            this.loadHistoricData(this.maxPoints, false, 86400);
+            break;
+          case 'customDate':
+            this.flushData();
+            if (changes.maxPoints && this.clickedOk) {
+              this.loadHistoricData(this.maxPoints, true);
+            }
+            break;
+          default:
+            break;
+        }
+      }
+      if (changes.startDate || changes.endDate) {
+        this.flushData();
+      }
+      if (changes.clickedOk) {
+        if (this.clickedOk) {
+          switch (this.options) {
+            case 'customDate':
+              this.loadHistoricData(this.maxPoints, true);
+              break;
+            default:
+              break;
+          }
+        }
+      }
+    } else {
+      this.initialized = true;
+      this.loadNewData();
+    }
+  }
+
+  public loadHistoricData(maxPoints: number, useDate: boolean, secondsInPast?: number): void {
+    if (useDate) {
+      const startDate = this.startDate.valueOf();
+      const endDate = moment(this.endDate).add(1, 'days').valueOf();
+      this.latestPoints$ = this.oispService.getValuesOfSingleFieldByDates(this.asset, this.fieldDetails, startDate, endDate, maxPoints);
+    } else {
+      this.latestPoints$ = this.oispService.getValuesOfSingleField(this.asset, this.fieldDetails, secondsInPast, maxPoints);
+    }
+    this.latestPoints$.pipe(takeUntil(this.destroy$))
+      .subscribe(
+        points => {
+          this.updateChart(points);
+          this.initLineChartOptions();
+        }
+      );
+  }
+
+  public updateChart(points: PointWithId[]): void {
+    points.forEach(point => {
+      this.currentNumberOfPoints += 1;
+      const data = this.lineChartData[0].data as ChartPoint[];
+      const dateOfPoint: moment.Moment = moment(point.ts);
+      data.push({ y: point.value, t: dateOfPoint });
+      this.lineChartLabels.push(dateOfPoint.toISOString());
+    });
+
+    if (this.isStatus) {
+      this.updateStatusPoints();
+    } else {
+      this.updateThresholds(points);
+      this.chart.update();
+    }
+
+    this.loadedEvent.emit();
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next(true);
+    this.destroy$.complete();
+  }
+
   private initLineChartOptions(): void {
-    const minMaxYAxis =  AssetChartHelper.getYMinMaxByAbsoluteThreshold(this.fieldDetails);
+    const minMaxYAxis = AssetChartHelper.getYMinMaxByAbsoluteThreshold(this.fieldDetails);
+
+    let xAxisStepSize = 6;
+    let xAxisUnit: TimeUnit = 'hour';
+    let minDate = AssetChartHelper.getMinDateForLineChart(this.lineChartData[0].data as ChartPoint[], null);
+    const maxDate = AssetChartHelper.getMaxDateForLineChart(this.lineChartData[0].data as ChartPoint[]);
+
+    const dayInMillisecs = 24 * 60 * 60 * 1000;
+    if (maxDate - minDate > 5 * dayInMillisecs) {
```

Comment:
I guess 5 is assessed by viewing data right?

Reply:
yes, just a tradeoff. When having data for more than 5 days, it seems more reasonable to have axis ticks in a 24h interval at minimum, while for less than 5 days 6h at minimum is more suitable.